### DISCUSSION TITLE
Export host - Flight Control

### DISCUSSION THREAD

**Discussion by waschoo** [CrowdStrike/psfalcon#202] [Q&A] [answered]
Hello,
I just move to flight control.
I used a script before to export all hosts from each child tenants (Get-FalconHost ....) in csv format. But I had to create an API for each child tenant.
Now I'd like to find a way to do it directly on the master tenant, I created an API key (host: ReadOnly), but  I'm not able to export machines that are stored in child tenant (result empty), there is a way to do this or I need to specify each tenant like the example below
https://github.com/CrowdStrike/psfalcon/wiki/Basic-Scripts#get-host-information-from-multiple-falcon-instances
Thanks for your help :)

**Comment by bk-cs** [comment, marked as answer]
> Hello, I just move to flight control. I used a script before to export all hosts from each child tenants (Get-FalconHost ....) in csv format. But I had to create an API for each child tenant. Now I'd like to find a way to do it directly on the master tenant, I created an API key (host: ReadOnly), but I'm not able to export machines that are stored in child tenant (result empty), there is a way to do this or I need to specify each tenant like the example below https://github.com/CrowdStrike/psfalcon/wiki/Basic-Scripts#get-host-information-from-multiple-falcon-instances Thanks for your help :)

You'll want to follow this example instead: https://github.com/CrowdStrike/psfalcon/wiki/Code-Examples#authorize-and-run-commands-in-member-cids

The script you used it meant for multiple standalone Falcon instances, not ones in a Parent/Child (Flight Control) configuration. In Flight Control, you can make an API Client in the parent which has access to each of the child CIDs. The host information of the children is not visible within the parent itself--you have to authenticate with each child and pull it that way. The example script is designed to authenticate with the parent, then get a list of the associated children and authenticate with each of those directly in order to run whatever code you'd like.

Note that exporting to CSV is not a perfect solution and may lead to data loss. If you want to ensure integrity of the device properties, exporting to Json is a better solution.
